crates/tui/src/config_tab.rs

```diff
@@ -15,8 +15,8 @@ use crate::{config::Config, HandleEvent};
 pub struct ConfigTab {
     config: Config,
     config_list_state: Mutex<ListState>,
-    origin_connections_collapsed: bool,
-    labels_collapsed: bool,
+    origin_connections_collapsed: Cell<bool>,
+    labels_collapsed: Cell<bool>,
 }
```

Comment:
In general, also not sure about this, as it seems that it is more likely that the struct contents here should be the discrete types themselves, and then the use of ConfigTab should be behind an Arc<Mutex<ConfigTab>>. This would be more idiomatic I think.